hi-ml/src/health_ml/deep_learning_config.py

```diff
@@ -0,0 +1,722 @@
+
+        if is_offline_run or output_to:
+            if output_to:
+                logging.info(f"All results will be written to the specified output folder {output_to}")
+                root = Path(output_to).absolute()
+            else:
+                logging.info("All results will be written to a subfolder of the project root folder.")
+                root = project_root.absolute() / DEFAULT_AML_UPLOAD_DIR
+            timestamp = create_unique_timestamp_id()
+            run_folder = root / f"{timestamp}_{model_name}"
+            outputs_folder = run_folder
+            logs_folder = run_folder / DEFAULT_LOGS_DIR_NAME
+        else:
+            logging.info("Running inside AzureML.")
+            logging.info("All results will be written to a subfolder of the project root folder.")
+            run_folder = project_root
+            outputs_folder = project_root / DEFAULT_AML_UPLOAD_DIR
+            logs_folder = project_root / DEFAULT_LOGS_DIR_NAME
+        logging.info(f"Run outputs folder: {outputs_folder}")
+        logging.info(f"Logs folder: {logs_folder}")
+        return DeepLearningFileSystemConfig(
+            outputs_folder=outputs_folder,
+            logs_folder=logs_folder,
+            project_root=project_root,
+            run_folder=run_folder
+        )
+
+    def add_subfolder(self, subfolder: str) -> DeepLearningFileSystemConfig:
+        """
+        Creates a new output folder configuration, where both outputs and logs go into the given subfolder inside
+        the present outputs folder.
+
+        :param subfolder: The subfolder that should be created.
+        :return:
+        """
+        if self.run_folder:
+            outputs_folder = self.run_folder / subfolder
+            logs_folder = self.run_folder / subfolder / DEFAULT_LOGS_DIR_NAME
+            outputs_folder.mkdir(parents=True, exist_ok=True)
+            logs_folder.mkdir(parents=True, exist_ok=True)
+            return DeepLearningFileSystemConfig(
+                outputs_folder=outputs_folder,
+                logs_folder=logs_folder,
+                project_root=self.project_root
+            )
+        raise ValueError("This method should only be called for runs outside AzureML, when the logs folder is "
+                         "inside the outputs folder.")
+
+
+class WorkflowParams(param.Parameterized):
+    """
+    This class contains all parameters that affect how the whole training and testing workflow is executed.
+    """
+    random_seed: int = param.Integer(42, doc="The seed to use for all random number generators.")
+    number_of_cross_validation_splits: int = param.Integer(0, bounds=(0, None),
+                                                           doc="Number of cross validation splits for k-fold cross "
+                                                               "validation")
+    cross_validation_split_index: int = param.Integer(DEFAULT_CROSS_VALIDATION_SPLIT_INDEX, bounds=(-1, None),
```

Comment:
these field names are awkwardly long, and annoy me every time I use them on the commandline. crossval_count, crossval_index?